### PR TITLE
MODPUBSUB-31

### PR DESCRIPTION
**Pull request by Igor-Gorchakov**
Liquibase tool is a library for managing database changes and maintain relational database tables. It is chosen as an outcome of the spike MODPUBSUB-10.
Here we need to move changes from PoC (which is done in MODPUBSUB-10) into the real mod-pubsub module, that is in the folio-org code base.

### CODE REVIEW

**Comment by OleksiiKuzminov** [comment]
plz make it buildable

File: mod-pubsub-server/src/main/java/org/folio/dao/EventDao.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ */
+public interface EventDao {
+  /**
+   * Returns event by
+   *
+   * @param eventId  event id
+   * @param tenantId tenant id
+   * @return event entity
+   */
+  Future getById(String eventId, String tenantId);
```
Comment (OleksiiKuzminov) [inline mod-pubsub-server/src/main/java/org/folio/dao/EventDao.java]: parametrized future

File: mod-pubsub-server/src/main/java/org/folio/dao/impl/EventDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  private static final String TABLE_NAME = "events";
+  @Autowired
+  private PostgresClientFactory pgClientFactory;
+
+  @Override
+  public Future<Optional<Event>> getById(String eventId, String tenantId) {
+    Future<ResultSet> future = Future.future();
+    try {
+      String getByIdQuery = "SELECT * FROM %s.%s WHERE id = ?";
```
Comment (OleksiiKuzminov) [inline mod-pubsub-server/src/main/java/org/folio/dao/impl/EventDaoImpl.java]: move to constant plz

File: mod-pubsub-server/src/main/java/org/folio/dao/impl/EventDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+public class EventDaoImpl implements EventDao {
+
+  @Autowired
+  private PostgresClientFactory pgClientFactory; // NOSONAR
+
+  @Override
+  public Future<Optional<Event>> getById(String eventId, String tenantId) {
+    /*
+      Implementation is stubbed.
+      Use pgClientFactory to obtain PostgresClient instance and
```
Comment (Igor-Gorchakov) [inline mod-pubsub-server/src/main/java/org/folio/dao/impl/EventDaoImpl.java]:
```suggestion
      Use PostgresClientFactory to obtain PostgresClient instance and
```

File: mod-pubsub-server/src/main/java/org/folio/dao/util/LiquibaseUtil.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  private static final String CHANGELOG_MODULE_PATH = "liquibase/module/changelog.xml";
+  private static final String CHANGELOG_TENANT_PATH = "liquibase/tenant/changelog.xml";
+  private static final String MODULE_CONFIGURATION_SCHEMA = "pubsub_config";
+
+  private LiquibaseUtil() {}
+
+  /**
+   * Performs initialization for module configuration schema:
+   * - creates schema for module configuration
+   * - runs scripts to fill module schema in
```
Comment (Igor-Gorchakov) [inline mod-pubsub-server/src/main/java/org/folio/dao/util/LiquibaseUtil.java]:
```suggestion
   * - runs scripts to fill module schema 
```

File: mod-pubsub-server/src/main/java/org/folio/dao/util/SingleConnectionProvider.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+   * @return Connection to database
+   * @throws SQLException if error occurs while trying to obtain access to database
+   */
+  public static Connection getConnection(Vertx vertx, String tenant) throws SQLException {
+    JsonObject connectionConfig = PostgresClient.getInstance(vertx, tenant).getConnectionConfig();
+    return getConnectionInternal(connectionConfig);
+  }
+
+  /**
+   * Returns database connection for the public tenant
```
Comment (Igor-Gorchakov) [inline mod-pubsub-server/src/main/java/org/folio/dao/util/SingleConnectionProvider.java]:
```suggestion
   * Returns database connection for the current user
```

File: NEWS.md [hunk 6]
```diff
@@ -4,3 +4,6 @@
 * Changed project structure to contain server and client parts. Client builds as a lightweight java library
 * Extended Event schema
 * Added samples
+* Applied Liquibase scripting tool to manage database tables
```
Comment (marcjohnson-kint) [inline NEWS.md]: From my perspective, the news should inform prospective adopters or clients of the meaningful changes. I don't really follow how these advise an adopter or client about what has changed.

**Pull request merged**